src/components/with-validator/redux/reducer.ts

```diff
@@ -54,6 +57,21 @@ export default function reducer(
         .set('shapesCollection', null)
         .set('shapesError', fromJS(action.payload))
         .set('isFetchingShapes', false);
+    case FETCH_ONTOLOGY_COLLECTION_REQUESTED:
+      return state
+        .set('ontologyCollection', null)
+        .set('ontologyError', null)
+        .set('isFetchingOntologies', true);
+    case FETCH_ONTOLOGY_COLLECTION_SUCCEEDED:
+      return state
+        .set('ontologyCollection', fromJS(action.payload.ontologyCollection))
+        .set('ontologyError', null)
+        .set('isFetchingOntologies', false);
+    case FETCH_ONTOLOGY_COLLECTION_FAILED:
+      return state
+        .set('ontologyCollection', null)
+        .set('ontologyError', fromJS(action.payload))
+        .set('isFetchingOntologies', false);
```

Comment:
Ikke glem å definere initiell state også på linje 18

```suggestion
    case FETCH_ONTOLOGIES_REQUESTED:
      return state
        .set('ontologies', fromJS([]))
        .set('ontologyError', null)
        .set('isFetchingOntologies', true);
    case FETCH_ONTOLOGIES_SUCCEEDED:
      return state
        .set('ontologies', fromJS(action.payload.ontologies))
        .set('ontologyError', null)
        .set('isFetchingOntologies', false);
    case FETCH_ONTOLOGIES_FAILED:
      return state
        .set('ontologies', null)
        .set('ontologyError', fromJS(action.payload))
        .set('isFetchingOntologies', false);
```